ffyaml/ffyaml.go

```diff
@@ -48,29 +55,36 @@ func valsToStrs(val interface{}) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
-	return []string{s}, nil
+	return []*string{s}, nil
 
 }
 
-func valToStr(val interface{}) (string, error) {
+func valToStr(val interface{}) (*string, error) {
+	rtnStr := ""
+	var err error
+
 	switch v := val.(type) {
 	case byte:
-		return string([]byte{v}), nil
+		rtnStr = string([]byte{v})
 	case string:
-		return v, nil
+		rtnStr = v
 	case bool:
-		return strconv.FormatBool(v), nil
+		rtnStr = strconv.FormatBool(v)
 	case uint64:
-		return strconv.FormatUint(v, 10), nil
+		rtnStr = strconv.FormatUint(v, 10)
 	case int:
-		return strconv.Itoa(v), nil
+		rtnStr = strconv.Itoa(v)
 	case int64:
-		return strconv.FormatInt(v, 10), nil
+		rtnStr = strconv.FormatInt(v, 10)
 	case float64:
-		return strconv.FormatFloat(v, 'g', -1, 64), nil
+		rtnStr = strconv.FormatFloat(v, 'g', -1, 64)
+	case nil:
+		return nil, nil
 	default:
-		return "", ff.StringConversionError{Value: val}
+		return nil, ff.StringConversionError{Value: val}
 	}
+
+	return &rtnStr, err
```

Comment:
Why can't you return from each case block?

Reply:
> Why can't you return from each case block?

I struggled with this very question for a bit. 

From what I could tell, using the `&` operator to convert the string var to a pointer was not allowed in the same (in-line) operation as the function calls to `string()`, line 68 or `strconv.Format*` as on lines 72,74, etc. At least, I saw an example that said it was not correct syntax.